SYS_ATL/memory.py

```diff
@@ -120,7 +120,7 @@ def alloc(cls, new_name, prim_type, shape, srcinfo):
         if len(shape) == 0:
             return f"{prim_type} {new_name};"
 
-        return (f"{prim_type} *{new_name} = "
+        return (f"{prim_type} *{new_name} = ({prim_type} *)"
```

Comment:
This not needed... why was this added?

Reply:
@yamaguchi1024 - this is injected into generated code that should not be compiled with C++.

Reply:
Clang++ was more strict about the type and IIRC we needed that casting to compile with `clang++` so yeah this change is related to us needing to use C++ compiler, but I don't see a reason why explicit casting is a problem.

Reply:
It's because C and C++ are different languages and C++ is _not_ a superset of C. It's not that Clang++ is "more strict", rather Clang++ is following _different rules_. Compiling our generated code with a C++ compiler is explicitly not supported and ever doing so increases the risk of a language incompatibility bug that outright kills our portability argument.

Reply:
I agree that we shouldn't explicitly support C++. I can try with @skeqiqevian not to use C++ compiler (looking at the current code it seems like C `const` would work? not sure) so I think this change can be reverted, but I think the change to `LoopIR_compiler.py` need to stay as we need a way not to compile buffer as a pointer.

Reply:
> I think the change to LoopIR_compiler.py need to stay as we need a way not to compile buffer as a pointer.

I am not exactly clear what's happening here, either. Can you show me an example of the generated code? If this is equivalent to `(intptr){arg_data}`, then just use that...